### ISSUE TITLE
Variables are mixed up

### ISSUE THREAD

**Issue by ffried** [mikefarah/yq#1458]
**Describe the bug**
Variables seem to be mixed up when using within a `map`.

Version of yq: 4.30.5
Operating system: mac
Installed via: homebrew

**Input Yaml**
Concise yaml document(s) (as simple as possible to show the bug, please keep it to 10 lines or less)
input.yml:
```yaml
key1:
  - value1A
  - value1B
  - value1C

key2:
  - value2A
  - value2B
  - value2C
```

**Command**
The command you ran:
```
yq e '. as $root | keys | map(select(. == "key1" or . == "key2") | . as $kind | $root[$kind] | map({"combination": ($kind + "/" + .)})) | flatten' input.yml
```

**Actual behavior**

```yaml
- combination: key1/value1A
- combination: key2/value1A
- combination: key1/value1B
- combination: key2/value1B
- combination: key1/value1C
- combination: key2/value1C
- combination: key1/value2A
- combination: key2/value2A
- combination: key1/value2B
- combination: key2/value2B
- combination: key1/value2C
- combination: key2/value2C
```

**Expected behavior**

```yaml
- combination: key1/value1A
- combination: key1/value1B
- combination: key1/value1C
- combination: key2/value2A
- combination: key2/value2B
- combination: key2/value2C
```

**Additional context**

I'm aware the the expression isn't fully optimized (e.g. `pick` could be used). But the very same expression works in `jq`:

```
yq -o=json input.yml | jq '. as $root | keys | map(select(. == "key1" or . == "key2") | . as $kind | $root[$kind] | map({"combination": ($kind + "/" + .)})) | flatten' -
```

results in:

```json
[
  {
    "combination": "key1/value1A"
  },
  {
    "combination": "key1/value1B"
  },
  {
    "combination": "key1/value1C"
  },
  {
    "combination": "key2/value2A"
  },
  {
    "combination": "key2/value2B"
  },
  {
    "combination": "key2/value2C"
  }
]
```

which is the equivalent to the expected yaml.

**Comment by mikefarah** [comment]
Thanks for the detailed bug description, this will be fixed in the next release along with https://github.com/mikefarah/yq/issues/1566

**Comment by mikefarah** [comment]
Fixed in v4.31.2
